backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -462,7 +465,20 @@ public RepositoryEntity setRepository(
     session.getTransaction().commit();
     if (create) {
       try {
-        createRoleBindingsForRepository(repository, userInfo, repositoryEntity);
+        roleService.createWorkspacePermissions(
+            repository.getWorkspaceServiceId(),
+            Optional.of(repository.getWorkspaceType()),
+            String.valueOf(repository.getId()),
+            Optional.empty(), // UAC will populate the owner ID
+            repositoryEntity.isDataset()
+                ? ModelDBServiceResourceTypes.DATASET
+                : ModelDBServiceResourceTypes.REPOSITORY,
+            CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
+            repository.getVisibility());
+        LOGGER.debug("Project role bindings created successfully");
+        Transaction transaction = session.beginTransaction();
+        repositoryEntity.setCreated(true);
+        transaction.commit();
```

Comment:
what about updating repository visibility and default permissions?

Reply:
I hope it will be the same as the project.

Reply:
I think in repositories you can update visibility, in projects it seems that you can't